services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/DefaultS3TransferManager.java

```diff
@@ -460,10 +461,19 @@ private static void handleException(CompletableFuture<CompletedFileDownload> ret
                                         CompletableFuture<TransferProgress> progressFuture,
                                         CompletableFuture<DownloadFileRequest> newDownloadFileRequestFuture,
                                         Throwable throwable) {
-        SdkClientException exception = SdkClientException.create("Failed to resume the request", throwable);
-        returnFuture.completeExceptionally(exception);
-        progressFuture.completeExceptionally(exception);
-        newDownloadFileRequestFuture.completeExceptionally(exception);
+        Throwable cause = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+        if (cause instanceof SdkException || cause instanceof Error) {
```

Comment:
Can we have a test case for cause `instanceof SdkException` check ?
Why do we have this check here ? And in what cases we get `SdkException` ? Is it fine no to communicate           "Failed to resume the request" in the exception.

Reply:
Added additional test and attached failed message to SdkException